ssstar/src/custom_credentials_provider.rs

```diff
@@ -0,0 +1,85 @@
+//! This module provides functionality of custom credentials provider
+//!
+//! The credentials are automatically refreshed if the credentials are expired that allows to use
+//! the this credential provider for long running jobs
+
+use aws_credential_types::provider::error::CredentialsError;
+use aws_credential_types::{provider::ProvideCredentials, Credentials};
+use futures::future::BoxFuture;
+use std::str::FromStr;
+use std::{fmt::Debug, sync::Arc};
+
+/// This represents the credentials to access to AWS API.
+#[derive(Clone, Debug)]
+pub struct CustomCredentialsUpdateOutput {
+    pub access_key: String,
+    pub secret_key: String,
+    pub session_token: Option<String>,
+}
+
+/// This is a trait that represents a callback to update credentials when they become expired
+pub trait CustomCredentialsUpdateCallback: Debug + Send + Sync {
```

Comment:
Why not to use `#[async_trait]`?

Reply:
i just automatically written it without `async_trait`, it didn't look so bad to me. Re-written with `async_trait`.